integrations/asana/client.go

```diff
@@ -0,0 +1,116 @@
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type integration struct{ vars sdkservices.Vars }
+
+var (
+	patSecret     = sdktypes.NewSymbol("pat_secret")
+	authType      = sdktypes.NewSymbol("authType")
+	integrationID = sdktypes.NewIntegrationIDFromName("asana")
+)
+
+var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.IntegrationPB{
+	IntegrationId: integrationID.String(),
+	UniqueName:    "asana",
+	DisplayName:   "Asana",
+	Description:   "Asana is a web and mobile application designed to help teams organize, track, and manage their work.",
+	LogoUrl:       "/static/images/asana.svg",
+	UserLinks: map[string]string{
+		"Asana developer platform": "https://developers.asana.com/",
+	},
+	ConnectionUrl: "/asana/connect",
+	ConnectionCapabilities: &sdktypes.ConnectionCapabilitiesPB{
+		RequiresConnectionInit: true,
+	},
+}))
+
+func New(cvars sdkservices.Vars) sdkservices.Integration {
+	i := &integration{vars: cvars}
+	return sdkintegrations.NewIntegration(
+		desc,
+		sdkmodule.New(),
+		connStatus(i),
+		connTest(i),
+		sdkintegrations.WithConnectionConfigFromVars(cvars),
+	)
+}
+
+// connStatus is an optional connection status check provided by
+// the integration to AutoKitteh. The possible results are "Init
+// required" (the connection is not usable yet) and "Initialized".
+func connStatus(i *integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionStatus(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+
+		vs, err := i.vars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		at := vs.Get(authType)
+		if !at.IsValid() || at.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+
+		if at.Value() == integrations.Init {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "Initialized"), nil
+		}
+		return sdktypes.NewStatus(sdktypes.StatusCodeError, "Bad auth type"), nil
+	})
+}
+
+// connTest is an optional connection test provided by the integration
+// to AutoKitteh. It is used to verify that the connection is working
+// as expected. The possible results are "OK" and "error".
+func connTest(i *integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionTest(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+
+		vs, err := i.vars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		req, err := http.NewRequest("GET", "https://app.asana.com/api/1.0/users/me", nil)
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		ps := vs.Get(patSecret).Value()
+		req.Header.Add("Authorization", "Bearer "+ps)
+
+		client := &http.Client{}
```

Comment:
If you just use `DefaultClient` (a.k.a. `Client{}`), you might as well just use `http.Get()` without a client at all.